### PR TITLE
Raise a warning if @pytest.mark.asyncio is applied to non-async function

### PR DESCRIPTION
**Pull request by asvetlov**

### CODE REVIEW

File: pytest_asyncio/plugin.py [hunk 1]
```diff
@@ -424,34 +432,43 @@ def wrap_in_sync(func: Callable[..., Awaitable[Any]], _loop: asyncio.AbstractEve
...
-                if task.done() and not task.cancelled():
-                    task.exception()
-                raise
+        if not inspect.isawaitable(coro):
+            pyfuncitem.warn(
+                pytest.PytestWarning(
+                    f"The test {pyfuncitem} is marked with '@pytest.mark.asyncio' "
+                    "but it is not an async function. "
+                    "Please remove asyncio marker, "
+                    "use 'asyncio_mode = auto' for autocollection."
```
Comment (seifertm) [inline pytest_asyncio/plugin.py:441]: It makes a lot of sense to issue a warning when the user marks a synchronous function as `asyncio`.

As a user, I would be confused by the instruction to use `asyncio_mode=auto`, though. What does that have to do with the misuse of `pytest.mark.asyncio`?

Comment (asvetlov) [reply pytest_asyncio/plugin.py:441]: I'm not sure about the best error text. Please feel free to correct it.
My thought was that the global mark (`pytestmark = pytest.mark.asyncio`) can lead to false markers easily.
Maybe we don't need to emphasize it. `"Please remove asyncio marker."` without mention `asyncio_mode = auto` is an option.
What do you think?

Comment (seifertm) [reply pytest_asyncio/plugin.py:441]: Ah, of course! The mark can also be applied when the user writes `pytestmark = pytest.mark.asyncio`. I didn't think of that.

I would definitely mention pytestmark, but leave out `asyncio_mode`. Maybe something like this?

"The test {pyfuncitem} is marked with '@pytest.mark.asyncio'  but it is not an async function. "
"Please remove the asyncio marker. If the test is not marked explicitly, check for global markers applied via 'pytestmark'"

Comment (asvetlov) [reply pytest_asyncio/plugin.py]: Sounds good to me. Updated the text.

**Review by seifertm — APPROVED** [review]

**Pull request merged**